cmd/kubeadm/app/apis/kubeadm/validation/validation_test.go

```diff
@@ -458,3 +475,158 @@ func TestValidateFeatureGates(t *testing.T) {
 		}
 	}
 }
+
+func TestValidateArgSelection(t *testing.T) {
+	var tests = []struct {
+		name     string
+		c        *kubeadm.NodeConfiguration
+		expected bool
+	}{
+		{
+			"invalid: DiscoveryToken and DiscoveryFile cannot both be set",
+			&kubeadm.NodeConfiguration{
+				DiscoveryFile:  "https://url/file.conf",
+				DiscoveryToken: "abcdef.1234567890123456@foobar",
+			},
+			false,
+		},
+		{
+			"invalid: DiscoveryToken or DiscoveryFile must be set",
+			&kubeadm.NodeConfiguration{
+				DiscoveryFile:  "",
+				DiscoveryToken: "",
+			},
+			false,
+		},
+		{
+			"invalid: DiscoveryTokenAPIServers not set",
+			&kubeadm.NodeConfiguration{
+				DiscoveryToken: "abcdef.1234567890123456@foobar",
+			},
+			false,
+		},
+		{
+			"invalid: DiscoveryTokenCACertHashes cannot be used with DiscoveryFile",
+			&kubeadm.NodeConfiguration{
+				DiscoveryFile:              "https://url/file.conf",
+				DiscoveryTokenCACertHashes: []string{"sha256:7173b809ca12ec5dee4506cd86be934c4596dd234ee82c0662eac04a8c2c71dc"},
+			},
+			false,
+		},
+		{
+			"WARNING: using token-based discovery without DiscoveryTokenCACertHashes can be unsafe",
+			&kubeadm.NodeConfiguration{
+				DiscoveryToken:                         "abcdef.1234567890123456@foobar",
+				DiscoveryTokenUnsafeSkipCAVerification: false,
+				DiscoveryTokenAPIServers:               []string{"192.168.122.100:9898"},
```

Comment:
please remove all references to 9898 in this file. That's a legacy from v1.4-1.5 and should not be present anymore
The port should be 6443

Reply:
done.